# json_query: extend list of type aliases for compatibility with ansible-core 2.19
- https://github.com/ansible-collections/community.general/pull/10539

Index: ansible_collections/community/general/tests/integration/targets/filter_json_query/tasks/main.yml
--- ansible_collections/community/general/tests/integration/targets/filter_json_query/tasks/main.yml.orig
+++ ansible_collections/community/general/tests/integration/targets/filter_json_query/tasks/main.yml
@@ -11,4 +11,23 @@
 - name: Test json_query filter
   assert:
     that:
-      - "users | community.general.json_query('[*].hosts[].host') == ['host_a', 'host_b', 'host_c', 'host_d']"
+      - >-
+        users | community.general.json_query('[*].hosts[].host') == ['host_a', 'host_b', 'host_c', 'host_d']
+      - >-
+        ports | json_query("[?contains(ports, `22`)]") == [ports[0]]
+      - >-
+        ports | json_query("[?contains(rule_desc, `ssh`)]") == [ports[0]]
+      - >-
+        my_complex_data | json_query('users[?id==`1`]') == [my_complex_data['users'][0]]
+  vars:
+    my_complex_data:
+      users:
+        - id: 1
+          name: Alice
+          roles: ["admin", "dev"]
+          status: active
+    ports:
+      - ports: [22]
+        rule_desc: "ssh"
+      - ports: [80]
+        rule_desc: "http"
